hack/setup-go.sh

```diff
@@ -0,0 +1,17 @@
+# Returns a GOPATH local to the source tree
```

Comment:
This script will be invoked for any make target, and will re-create the go path every time. Can we move all this logic into the Makefile itself to avoid this? Similar to how `hellogopher` does it: https://github.com/cloudflare/hellogopher/blob/master/Makefile#L101-L107